tests/test_utils.py

```diff
@@ -126,18 +145,18 @@ def test_message_list_reduced(self):
         """
         # pass in copy of list, as it is modified as part of the reduction
         reduced_message_list = get_reduced_message_list(self.prompt_template, self.message_list)
-        self.assertEqual(len(reduced_message_list), 2)
+        self.assertEqual(len(reduced_message_list), 1)
```

Comment:
Question: might be missing this in the implementation but can you clarify why the length goes down by 1 for this assert and the assert below?